src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -81,16 +56,51 @@ public static bool IsRunningUnderIDE
 
         #region Protected Helper Methods
 
+        // The Adapter is constructed using the default constructor.
+        // We don't have any info to initialize it until one of the
+        // ITestDiscovery or ITestExecutor methods is called. The
+        // each Discover or Execute method must call this method.
+        protected virtual void Initialize(IMessageLogger messageLogger)
+        {
+            _verbosity = 0; // In case we throw below
+            _adapterVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+
+            try
+            {
+                var registry = RegistryCurrentUser.OpenRegistryCurrentUser(@"Software\nunit.org\VSAdapter");
+                UseVsKeepEngineRunning = registry.Exist("UseVsKeepEngineRunning") && (registry.Read<int>("UseVsKeepEngineRunning") == 1);
+                _shadowCopy = registry.Exist("ShadowCopy") && (registry.Read<int>("ShadowCopy") == 1);
```

Comment:
We had a discussion about this somewhere. The decision was that shadow-copy would be false by default but that individual runners would change it as necessary. AFAIK, only the NUnit gui will need to use it.

I agree about settings. Use of the registry is inconsistent with the rest of NUnit. Note, however, that there is not curently any easy-to-use gui for editing the NUnit settings - you have to edit the xml file directly. @OsirisTerje what do you think?

Further note on settings: both the registry and the NUnit xml settings are suitable for defaults. We really need to add a way for folks to change some things on a run-by-run basis.